chaintesting/action_test_helpers.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintesting
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+var _ state.Mutable = (*inMemoryStore)(nil)
+
+// inMemoryStore is a storage that acts as a wrapper around a map and implements state.Mutable.
+type inMemoryStore struct {
+	Storage map[string][]byte
+}
+
+func NewInMemoryStore() *inMemoryStore {
```

Comment:
I think it's return to return a non-exported type from a exported function. We should either return the interface or just export `inMemoryStore`. I recommend the latter so that the caller can decide to use it as a concrete type or as an interface

Reply:
The point of the `NewInMemoryStore` was to be a constructor for the `inMemoryStore` struct, I don't understand why the value associated with the `inMemoryStore` could not be used as an interface, for instance we can currently call `SetBalance` with the parameter `mu state.Mutable` as type without any issue

Reply:
This compiles fine now because you're just using it as a variable:

```
db := NewInMemoryStore()
db.Put()
db.Get()
...
```

should compile. But because the concrete-type isn't exported outside of the package a caller can't do this for example:

```
type Foo struct {
    db *chaintest.inMemoryStore
}
```

Because that type isn't exported outside of the `chaintest` package so other packages don't know that it type is referring to. If we're going to return a type outside of the package via `NewInMemoryStore`, I think we should allow callers to know what that concrete type is. This go pattern is considered non-idiomatic (go linters will warn you with "Exported function with the unexported return type", my IDE does this automatically).

Reply:
I see ! Thanks a lot for the explanation !

Reply:
In a way, returning the unexported type is an advantage.
It forces the caller to either only use exported methods on the returned type, or declare an explicit interface on their side (caller side), which is the correct side for the dependency to live.
This makes it very easy to switch in a different callee package or to return a wrapped type later, since it avoids embedding types by name (either struct or interfaces) in the user (caller)'s code.